src/native/common/include/runtime-base/strings.hh

```diff
@@ -743,6 +743,16 @@ namespace xamarin::android {
 			return starts_with (s.data (), s.length ());
 		}
 
+		[[gnu::always_inline]]
+		auto ends_with (std::string_view const& s) noexcept -> bool
+		{
+			if (empty () || s.length () > buffer.size ()) {
+				return false;
+			}
+
+			return memcmp (buffer.get () + (length () - s.length ()), s.data (), s.length ()) == 0;
```

Comment:
Just for me to understand, `length()` is always inlined, so no problem with calling it multiple times in an expression?

https://github.com/dotnet/android/blob/2ef003801d38c72be0eb73972ad8dc1f73bfa498/src/native/common/include/runtime-base/strings.hh#L452-L453

Reply:
Yeah, it's fully inlined each time it's called (which can also the be further optimized by common sub-expression elimination and other techniques, depending on context)